toolchain/parse/handle_expr.cpp

```diff
@@ -401,14 +401,9 @@ auto HandleExprStatementFinish(Context& context) -> void {
     context.emitter().Emit(*context.position(), ExpectedExprSemi);
   }
 
-  if (auto semi_token = context.SkipPastLikelyEnd(state.token)) {
-    context.AddNode(NodeKind::ExprStatement, *semi_token, state.subtree_start,
-                    /*has_error=*/true);
-    return;
-  }
-
-  // Found junk not even followed by a `;`, no node to add.
-  context.ReturnErrorOnState();
+  auto last_consumed_token = context.SkipPastLikelyEnd(state.token);
+  context.AddNode(NodeKind::ExprStatement, last_consumed_token,
```

Comment:
```suggestion
  context.AddNode(NodeKind::ExprStatement, context.SkipPastLikelyEnd(state.token),
```

Maybe inline the variable in cases like this (also in other files)